lib/src/file.dart

```diff
@@ -398,9 +398,8 @@ class _FileSpan extends SourceSpanMixin implements FileSpan {
         sourceUrl == other.sourceUrl;
   }
 
-  // Eliminates dart2js warning about overriding `==`, but not `hashCode`
   @override
-  int get hashCode => super.hashCode;
+  int get hashCode => Object.hash(_start, _end, sourceUrl);
```

Comment:
This looks like it was intentionally using the identity hash code? I would probably do some digging at least to understand why it was done the way it was.

Reply:
It's not identity – it's the hash from the parent.

I wanted to avoid the warning.

I can put it back

Reply:
I just assumed it was falling back on `Object.hashCode` so its identity - if its actually hitting some real super implementation that would be even more likely to be an intentional thing I would think - maybe the identity isn't intended to take these other fields into account etc.

Reply:
Just to be clear in general we should probably have a real hash code implementation here, but somebody very intentionally didn't do that so I wouldn't want to change it without understanding why.

Reply:
We may need a time machine and ask @kevmoo (see https://github.com/dart-lang/source_span/commit/3cf8669673b639b164e5f46fd358983051d15b0c)

Reply:
Hah!

Reply:
Actually given that I think its probably safe to just add the real implementation @kevmoo .

Reply:
Yeah the real implementation looks like the best choice.